pkg/vm/context.go

```diff
@@ -171,8 +171,7 @@ func (c *Context) Next() (opcode.Opcode, []byte, error) {
 	if err != nil {
 		return instr, nil, err
 	}
-	parameter := make([]byte, numtoread)
-	copy(parameter, c.prog[c.nextip:c.nextip+numtoread])
+	parameter := c.prog[c.nextip : c.nextip+numtoread]
```

Comment:
This reminds me of golang/go#27975, but until we have it we need to trace what exactly is happening to this `parameter`.

We have it being used in compiler/test code, but it's not dangerous there.

`ParseSignatureContract` and `ParseMultiSigContract` return it, but all current users are safe, they only read these values.

`PrintOps` also reads things.

And then we have `v.execute` with a hell a lot of options. It's used for `GetPrice`, but it doesn't care. It's used to push integers onto the stack, but that inevitably leads to `bigint.FromBytes`, so not a problem. It's used for `PUSHDATA`, but we'll get to it later. Then `PUSHA`, `ISTYPE`, `CONVERT`, slot-related opcodes, `NEW*`, `JMP*`, `CALL*`, and `*TRY*` all read their parameters from it and it's all fine.

The most problematic case is `PUSHDATA`, because that's where a `ByteArray` item is created containing reference to this data. By design `ByteArray` is an immutable type (kinda like `string`), unlike `Buffer`, but we still need to check it. 

We can use it in `MEMCPY`, but only as a source that we copy data from. `CAT` can take it, but it also copies data from both of its parameters. `SUBSTR`, `LEFT` and `RIGHT` copy too. `PICKITEM` reads one byte and `SIZE` takes a `len()`. It can also be used in exception-handling code, but it's converted to `string` there immediately.

We have conversions though. Conversions to `ByteArray` don't copy data, but it's `ByteArray`, so it should be safe, `Buffer` does `slice.Copy` and `Integer` and `Bool` return new respective values. `Equals` doesn't change anything.

So, we're safe?

Reply:
Anyway, I think it's worth mentioning in `Next` documentation that this `parameter` points to script data and user is responsible for copying it if needed.